pkg/auth/providers/common/usermanager.go

```diff
@@ -303,6 +313,58 @@ func (m *userManager) EnsureUser(principalName, displayName string) (*v3.User, e
 	return user, nil
 }
 
+func (m *userManager) CreateNewUserClusterRoleBinding(userName string, userUID apitypes.UID) error {
+	roleName := userName + "-view"
+	bindingName := "grb-" + roleName
+
+	ownerReference := v1.OwnerReference{
+		APIVersion: "management.cattle.io/v3",
+		Kind:       "User",
+		Name:       userName,
+		UID:        userUID,
+	}
+
+	rule := k8srbacv1.PolicyRule{
+		Verbs:         []string{"get"},
+		APIGroups:     []string{"management.cattle.io"},
+		Resources:     []string{"users"},
+		ResourceNames: []string{userName},
+	}
+	role := &k8srbacv1.ClusterRole{
+		ObjectMeta: v1.ObjectMeta{
+			Name:            roleName,
+			OwnerReferences: []v1.OwnerReference{ownerReference},
+		},
+		Rules: []k8srbacv1.PolicyRule{rule},
+	}
+
+	cr, err := m.RBAC.ClusterRoles("").Create(role)
+	if err != nil {
+		return err
+	}
+
+	crb := &k8srbacv1.ClusterRoleBinding{
+		ObjectMeta: v1.ObjectMeta{
+			Name:            bindingName,
+			OwnerReferences: []v1.OwnerReference{ownerReference},
+		},
+		Subjects: []k8srbacv1.Subject{
+			k8srbacv1.Subject{
+				Kind: "User",
+				Name: userName,
+			},
+		},
+		RoleRef: k8srbacv1.RoleRef{
+			Kind: "ClusterRole",
+			Name: cr.Name,
+		},
+	}
+
+	_, err = m.RBAC.ClusterRoleBindings("").Create(crb)
```

Comment:
should probably do a check to see if this already exists at the top of the function and just short circuit